### ISSUE TITLE
Seems to hang on certain audio streaming sites.

### ISSUE THREAD

**Issue by mbsmith** [qutebrowser/qutebrowser#7982]
**Version info**:

qutebrowser v3.0.2
Git commit: 
Backend: QtWebEngine 6.6, based on Chromium 112.0.5615.213 (from api)
Qt: 6.6.0

CPython: 3.11.5
PyQt: 6.5.2

Qt wrapper info:
  PyQt6: success
  PyQt5: not imported
  -> selected: PyQt6 (via autoselect)

**Does the bug happen if you start with `--temp-basedir`?**:
yes

**Description**
When streaming music from either youtube music, or apple music the browser freezes.  This includes
other qutebrowser windows.  Also upon restart if it reopens tabs they all show up blank and such so must be
restarted fresh.

**How to reproduce**
1. Go to music.youtube.com or music.apple.com
2. Play something, maybe navigate a little bit.  It happens within minutes for me, but I haven't pinned down the exact action that causes it.
3. Qutebrowser windows go non-responsive.

**Comment by mschilli87** [comment]
I have observed latest qutebrowser on Mac to freeze while playing YouTube videos as well. Could be related. I also have not managed to reproduce it reliably or figure out what's happening. Memory & CPU usage are fine.

**Comment by The-Compiler** [comment]
@mbsmith What OS/platform are you on?

**Comment by mbsmith** [comment]
Platform plugin: wayland
OpenGL: AMD, 4.6 (Compatibility Profile) Mesa 23.1.9
Platform: Linux-6.1.60-x86_64-with-glibc2.38, 64bit
Linux distribution: NixOS 23.11 (Tapir) (nixos)

**Comment by mschilli87** [comment]
Could this be a codec issue? AFAIR the MacOS (binary) build come without propriatary codec support and so might be the NixOS build...

**Comment by cortsf** [comment]
It also hangs here and from time to time (between 2 and 5 times per day) it also crashes. When it hangs I can scroll or see the bg effect while hovering on some (html..) items, but I can't select text, play videos or open the mouse menu. If i do right click, when it de-hangs (either by waiting ~30 secods or by switching to another tab) many menues (equally in number to the number of clicks, it seems) do open.

This happens very often on youtube and every single time when I search for packages on github using the search engine abreviation/alias/keyword instead of github's search bar. On both sites the UI shows it's  loading something when this happens. 

Using NixOS and X. QB being pulled from the "unstable" nixos channel.


```
qutebrowser v3.0.2
Git commit: 
Backend: QtWebEngine 6.6, based on Chromium 112.0.5615.213 (from api)
Qt: 6.6.0

CPython: 3.11.5
PyQt: 6.5.2

Qt wrapper info:
  PyQt6: success
  PyQt5: not imported
  -> selected: PyQt6 (via autoselect)
  ```

**Comment by cortsf** [comment]
I think it's fixed here after updating nixos by removing the `flake.lock` and rebuilding (by doing this I got QB broken a few weeks ago..). QB version is still the same, I think it didn't got updated in nixpkgs at all but maybe something got fixed elsewhere in the system. I didn't had the opportunity to test it thoroughly but it seems qutebrowser was also de-hanging just by switching to another xmonad workspace (Or by pressing any key, I don't know, maybe IO related (?)).

In turn, I'm having a lot of random `ERR_ADDRESS_UNREACHABLE` errors but I hope it's something temporal related to my connection and not qb or Nixos related.
